scripts/ansible-todd-dev.yml

```diff
@@ -1,83 +1,135 @@
+      stat:
+        path: "{{ todd_make_configureenv_check_file }}"
+      register: "make_configureenv_check"
+
+    - name: Configure Environment
+      command: "make configureenv" #what is changed in order to determine if this needs to run again
+      args:
+        chdir: "{{ todd_dir }}"
+      register: "make_configureenv"
+      when: not make_configureenv_check.stat.exists
+
+    - name: Marking Configure Environment As Complete
+      file:
+        path: "{{ todd_make_configureenv_check_file }}"
+        state: "touch"
+      when: make_configureenv.changed
+
+    - name: GOPATH Owner  #Needs to be changed to ensure idempotency
+#      file:
+#        path: "/home/vagrant/go"
+#        owner: "vagrant"
+#        group: "vagrant"
+#        recurse: yes
+#        state: "directory"
+      shell: chown vagrant:vagrant $(find /home/vagrant/go | grep -v todd) warn=no
 
 - name: Prepping for ToDD
   hosts: all
   vars:
-    todd_dir: /home/vagrant/go/src/github.com/Mierdin/todd
+    todd_compile_check_file: '/var/log/.todd_compile_complete'
+    todd_dir: '/home/vagrant/go/src/github.com/Mierdin/todd'
   tasks:
 
-  - name: Set GOPATH for Vagrant user profile
-    lineinfile: 
-      dest=/home/vagrant/.profile
-      state=present
-      regexp='^export GOPATH'
-      line='export GOPATH=/home/vagrant/go'
-
-  - name: Set PATH for Vagrant user profile
-    lineinfile: 
-      dest=/home/vagrant/.profile
-      state=present
-      regexp='^export PATH'
-      line='export PATH=$PATH:/home/vagrant/go/bin:/usr/local/go/bin:{{ todd_dir }}/scripts'
-
-  - name: cd to todd_dir when logging in
-    lineinfile: 
-      dest=/home/vagrant/.profile
-      state=present
-      regexp='^cd '
-      line='cd {{ todd_dir }}'
-
-  - name: Compile ToDD
-    shell: . /home/vagrant/.profile && make chdir={{ todd_dir }}
+    - name: Set GOPATH for Vagrant user profile
+      lineinfile:
+        dest: "/home/vagrant/.profile"
+        state: "present"
+        regexp: "^export GOPATH"
+        line: "export GOPATH=/home/vagrant/go"
+
+    - name: Set PATH for Vagrant user profile
+      lineinfile:
+        dest: "/home/vagrant/.profile"
+        state: "present"
+        regexp: "^export PATH"
+        line: "export PATH=$PATH:/home/vagrant/go/bin:/usr/local/go/bin:{{ todd_dir }}/scripts"
+
+    - name: cd to todd_dir when logging in
+      lineinfile:
+        dest: "/home/vagrant/.profile"
+        state: "present"
+        regexp: "^cd "
+        line: "cd {{ todd_dir }}"
+
+    - name: Checking Status of Compile ToDD
+      stat:
+        path: "{{ todd_compile_check_file }}"
+      register: "todd_compile_complete"
+
+    - name: Compile ToDD
+      shell: ". /home/vagrant/.profile && make"
+      args:
+        chdir: "{{ todd_dir }}"
+      register: "todd_compiled"
+      when: not todd_compile_complete.stat.exists
+
+    - name: Marking Status of Compile ToDD as Complete
+      file:
+        path: "{{ todd_compile_check_file }}"
```

Comment:
Getting a permissions error since it's trying to write directly to ```/var/log``` at this point:

```
TASK [Marking Status of Compile ToDD as Complete] ******************************
fatal: [todddev]: FAILED! => {"changed": false, "failed": true, "module_stderr": "", "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1463635546.78-15744541496339/file\", line 2405, in <module>\r\n    main()\r\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1463635546.78-15744541496339/file\", line 395, in main\r\n    open(path, 'w').close()\r\nIOError: [Errno 13] Permission denied: '/var/log/.todd_compile_complete'\r\n", "msg": "MODULE FAILURE", "parsed": false}
```

Maybe a different directory? Home dir seems a good a place as any for markers like this.